sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared_async.py

```diff
@@ -28,7 +28,7 @@ def __init__(self, response, default_http_response_type: Any) -> None:
         self._parts = None
         self._default_http_response_type = default_http_response_type
 
-    async def _parse_response(self):
+    async def _parse_response(self) -> Iterator:
```

Comment:
Will this be iter of `HttpResponse` ?

```suggestion
    async def _parse_response(self) -> Iterator[HttpResponse]:
```

Reply:
No, I don't know the concrete type at this point, it would be of type `default_http_response_type` that is in the client parameter. It can be the old transport type, or the new transport type, this file doesn't know.

I can try to make the `_PartGenerator` a generic, and see what is happening.

Reply:
Ok, got it to work with Generic :)

![image](https://github.com/Azure/azure-sdk-for-python/assets/1050156/4b8ffc47-5417-497c-a077-558e1f2db778)